### PR TITLE
Add bean soaking task

### PR DESCRIPTION
**Pull request by AkagiShigeru**
…s not technically necessary according to PCRE but does not hurt here).

### CODE REVIEW

**Comment by ArBridgeman** [comment]
@AkagiShigeru your changes looked good, so i approved them.
before merging, please see the additional changes we discussed for your branch that i added.

if it's not too much effort & you would like to, you could try to simplify the "separately scheduling" text. it's slightly more involved as i previously said as we don't have enums already available to do the short text on. we could see if we could modify the existing enum or a method of it to support such a short case.

alternately, we could also reduce the options available for preparation time, so instead of breakfast, lunch, dinner, or dessert, we could put [l]unch or [d]essert since we avoid dinner due to already working & breakfast since we don't want to cook that early.

**Review by ArBridgeman — APPROVED** [review]

File: sous-chef/sous_chef/grocery_list/generate_grocery_list.py [hunk 1]
```diff
@@ -271,25 +272,47 @@ def _add_referenced_recipe_to_queue(
...
             return response
 
         def _check_yes_no(text: str) -> str:
             response = None
             while response not in ["y", "n"]:
-                response = input(f"\n{text}[y]es, [n]o").lower()
+                response = input(f"\n{text} [y]es, [n]o: ").lower()
             return response
 
+        def print_enum(eprint: Enum) -> str:
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/grocery_list/generate_grocery_list.py:284]: nit: prefer `def _print_enum` as f(f); same with print_list

File: sous-chef/sous_chef/grocery_list/generate_grocery_list.py [hunk 2]
```diff
@@ -271,25 +272,47 @@ def _add_referenced_recipe_to_queue(
...
+            return ", ".join(
+                f"{lprint[il]} [{il}]" for il in range(len(lprint))
+            )
+
         def _get_schedule_day_hour_minute() -> datetime:
             day = None
-            while day not in Weekday.name_list("capitalize"):
-                day = input("\nWeekday: ").capitalize()
+            while (
+                not day or not day.isnumeric() or int(day) not in iter(Weekday)
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/grocery_list/generate_grocery_list.py:295]: do we want ORs or ANDs in these whiles?

Comment (ArBridgeman) [reply sous-chef/sous_chef/grocery_list/generate_grocery_list.py:295]: nvm forgot demorgans conversion; thanks for re-explaining 😊

File: sous-chef/sous_chef/grocery_list/generate_grocery_list.py (hunk 1 above)
Comment (AkagiShigeru) [reply sous-chef/sous_chef/grocery_list/generate_grocery_list.py:284]: done, thx

**Pull request merged**